### PR TITLE
[EdgeTPU] Extract Interface for Modularization (CfgEditorPanel)

### PR DESCRIPTION
**Pull request by hohee-hee**
This commit modularizes CfgEditorPanel for EdgeTPU Compiler. Two works are done in this commit.

- Create ICfgEditorPanel.ts : Interface for CfgEditorPanel and a new Panel for edgetpucfg
- Change CfgEditorPanel.ts :  Implements ICfgEditorPanel for .cfg (original config file)

ONE-vscode-DCO-1.0-Signed-off-by: hohee-hee <khappy517@gmail.com>

Co-authored-by: Bumsoo Ko <rhqjatn2398@naver.com>
Co-authored-by: Hyeon-Uk <rlagusdnr120@gmail.com>

### CODE REVIEW

File: src/CfgEditor/ICfgEditorPanel.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    initWebviewPanel(
+        document: vscode.TextDocument,
+        webviewPanel: vscode.WebviewPanel
+    ): void
+    
+    updateWebview(
+        document: vscode.TextDocument,
+        webview: vscode.Webview
+    ): void
+}
```
Comment (dayo09) [inline src/CfgEditor/ICfgEditorPanel.ts:25]: Could you add a new line at the end of the file?

File: src/CfgEditor/ICfgEditorPanel.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import * as vscode from "vscode";
+
+export interface ICfgEditorPanel extends vscode.CustomTextEditorProvider {
+
+    resolveCustomTextEditor(
```
Comment (dayo09) [inline src/CfgEditor/ICfgEditorPanel.ts:5]: It seems ICfgEditorPanel has no additional definitions or interface, so it's working just as an alias of vscode.CustomTextEditorProvider. Do you plan to add something here?

Comment (dayo09) [reply src/CfgEditor/ICfgEditorPanel.ts:5]: > So, in fact, during writting this comment, I think ICfgEditorPanel is useless file😓

I am not sure about your plan, so I wanna ask if you plan to modify this file or keep it..? :thinking:

Comment (hohee-hee) [reply src/CfgEditor/ICfgEditorPanel.ts:5]: Actually Nope.
There is no plan😕

Comment (dayo09) [reply src/CfgEditor/ICfgEditorPanel.ts:5]: If so, how about removing this one and using vscode.CustomTextEditorProvider, just like before?

Comment (hohee-hee) [reply src/CfgEditor/ICfgEditorPanel.ts:5]: I also think that's the better idea😊
Then, I will remove this code and make a new commit which includes `EdgeTPUCfgEditorPanel.ts` that implements vscode.CustomTextEditorProvider!

In this case, I plan to commit in this PR, but should I close this pull request and open a new one?

Comment (dayo09) [reply src/CfgEditor/ICfgEditorPanel.ts:5]: > In this case, I plan to commit in this PR, but should I close this pull request and open a new one?

You can choose whatever for your own convenience  :-D

Comment (hohee-hee) [reply src/CfgEditor/ICfgEditorPanel.ts:5]: Then I will make a new commit here, thanks!

**Comment by hohee-hee** [comment]
The implement part of this interface has other Object that is not merged at the ssafy/2023 branch yet.
So, this PR is closed and I will make a new PR after necessary code is merged.

**Pull request closed without merging**